core/src/main/java/software/amazon/awssdk/core/async/AsyncRequestProvider.java

```diff
@@ -87,4 +102,34 @@ static AsyncRequestProvider fromString(String string) {
         return fromString(string, StandardCharsets.UTF_8);
     }
 
+    /**
+     * Creates a {@link AsyncRequestProvider} from a byte array. The contents of the byte array are copied so modifications to the
+     * original byte array are not reflected in the {@link AsyncRequestProvider}.
+     *
+     * @param bytes The bytes to send to the service.
+     * @return AsyncRequestProvider instance.
+     */
+    static AsyncRequestProvider fromBytes(byte[] bytes) {
+        return new SingleByteArrayAsyncRequestProvider(Arrays.copyOf(bytes, bytes.length));
```

Comment:
Not in scope of this review but the name of this class is kinda confusing. Just ByteArrayAsyncRequestBody?